test/react/functional-components/simple-with-jsx-spread.js

```diff
@@ -0,0 +1,30 @@
+var React = require('react');
+// the JSX transform converts to React, so we need to add it back in
+this['React'] = React;
+
+function IWantThisToBeInlined(props) {
+  return <div>{props.text} {props.name} {props.id}</div>
+}
+
+function Button(props) {
+  return props.switch ? <IWantThisToBeInlined {...props} /> : null
+}
+
+Button.defaultProps = {
+	name: "Dominic",
+};
+
+function App(props) {
+  return <Button {...props} id={"5"} {...props} switch={true} />
```

Comment:
This test breaks if I put

```js
return <Button {...props} id={"5"} switch={true} />
```

instead. Is this a known issue or a bug?

Reply:
(On master it just fails, but at least it doesn't create wrong output. I think failing is always preferable. If there's some case we can't handle, could you bail out here?)

Reply:
It creates the wrong output because of the bugs on master with `Object.assign`.